### PR TITLE
Handle Query Updates

### PR DESCRIPTION
**Pull request by divyaac**
### Description
Change HandleQuery so as to return the updated current month estimation values.
Tests are available in the matching ENT PR. 

Matching ENT Pr: https://github.com/hashicorp/vault-enterprise/pull/6102

### TODO only if you're a HashiCorp employee
- [ ] **Labels:** If this PR is the CE portion of an ENT change, and that ENT change is
  getting backported to N-2, use the new style `backport/ent/x.x.x+ent` labels
  instead of the old style `backport/x.x.x` labels.
- [ ] **Labels:** If this PR is a CE only change, it can only be backported to N, so use
  the normal `backport/x.x.x` label (there should be only 1).
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

**Comment by miagilepner** [comment]
I was confused about why no tests were failing, before realizing that there's a bug in the implementation which is fixed here: https://github.com/hashicorp/vault/pull/27549.

**Review by miagilepner — COMMENTED** [review]
Can we also add a test to `vault/external_tests/activity_testonly/activity_testonly_test.go` to verify the response in the total field as well as the mount breakdowns?

**Comment by divyaac** [comment]
> activity_testonly_test.go

Added some tests to activity_log_test.go to verify some of these values (as I mentioned in the ent pr, this is a little iffy cause they're estimations. But i used a similar error margin approach.)

File: vault/external_tests/upgrade_testing/upgrade_testing_binary/upgrade_test.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/vault/sdk/helper/testcluster"
+	"github.com/hashicorp/vault/sdk/helper/testcluster/docker"
+)
+
+func DockerPrivilegedNamespaceClient(t *testing.T, nodeCount int) *docker.DockerCluster {
```
Comment (miagilepner) [inline vault/external_tests/upgrade_testing/upgrade_testing_binary/upgrade_test.go:11]: I think this wasn't meant to be committed

File: vault/activity_log_test.go [hunk 2]
```diff
@@ -5084,3 +5086,193 @@ func TestActivityLog_reportPrecomputedQueryMetrics(t *testing.T) {
...
+			startOfOneMonthAgo := timeutil.StartOfMonth(startOfMonth.AddDate(0, -1, 0)).UTC()
+			generatePreviousMonthClientData(t, core, startOfTwoMonthsAgo, entityRecordsMonth2)
+			generatePreviousMonthClientData(t, core, startOfOneMonthAgo, entityRecordsMonth1)
+
+			// Generate the current months data
+			clientPrefix += 1
+			var currentMonthData []*activity.EntityRecord
+			for namespaceId, mountSlice := range mounts {
+				for mountIndex, mount := range mountSlice {
+					currentMonthData = append(currentMonthData, generateClientData(0, tt.currentMonth[0][mountIndex], mount.Accessor, namespaceId, fmt.Sprintf("%d", clientPrefix))...)
```
Comment (miagilepner) [inline vault/activity_log_test.go:5238]: looks like generateClientData and generatePreviousMonthClientData are both defined in an enterprise file. Can we move them to CE?

Comment (divyaac) [reply vault/activity_log_test.go:5239]: Oops! Moved these over!

File: vault/external_tests/upgrade_testing/upgrade_testing_binary/upgrade_test.go (hunk 1 above)
Comment (divyaac) [reply vault/external_tests/upgrade_testing/upgrade_testing_binary/upgrade_test.go]: Oops again!! Removed this.

File: vault/activity_log_test.go [hunk 3]
```diff
@@ -5084,3 +5087,290 @@ func TestActivityLog_reportPrecomputedQueryMetrics(t *testing.T) {
...
+
+			// New verify that the new client totals for ALL namespaces are approximately accurate
+			newClientsError := math.Abs((float64)(currentMonthClients.NewClients.Counts.Clients - tt.expectedNewClients))
+			newClientsErrorMargin := newClientsError / (float64)(tt.expectedNewClients)
+			expectedAccuracyCalc := (1 - tt.expectedTotalAccuracy) * 100 / 100
+			if newClientsErrorMargin > expectedAccuracyCalc {
+				t.Fatalf("bad accuracy: expected %+v, found %+v", expectedAccuracyCalc, newClientsErrorMargin)
+			}
+
+			// Verify that the totals for the clients are visibly sensible (that is the total of all the individual new clients per namespace)
```
[the remark this answers is not in the document]
Comment (divyaac) [reply vault/activity_log_test.go:5269]: This part is done in the ent test in the ENT pr - this one is for the CE tests

File: vault/activity_log_test.go (hunk 2 above)
Comment (divyaac) [reply vault/activity_log_test.go]: refactored this to use clientcountutil

File: vault/activity_log_test.go [hunk 4]
```diff
@@ -5084,3 +5087,290 @@ func TestActivityLog_reportPrecomputedQueryMetrics(t *testing.T) {
...
+	tests := map[string]struct {
+		twoMonthsAgo          [][]int
+		oneMonthAgo           [][]int
+		currentMonth          [][]int
+		expectedNewClients    int
+		repeatPreviousMonth   bool
+		expectedTotalAccuracy float64
+	}{
+		"low volume, all mounts": {
+			twoMonthsAgo: [][]int{
```
[the remark this answers is not in the document]
Comment (divyaac) [reply vault/activity_log_test.go]: done

**Review by miagilepner — APPROVED** [review]
LGTM!!

**Pull request merged**